game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaOffenseCombatValue.java

```diff
@@ -0,0 +1,68 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+
+/**
+ * Calculates offense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Value
+@EqualsAndHashCode(callSuper = true)
+@Getter(AccessLevel.NONE)
+class AaOffenseCombatValue extends AaCombatValue {
+
+  @Builder
+  AaOffenseCombatValue(
+      final GameData gameData,
+      final AvailableSupports supportFromFriends,
+      final AvailableSupports supportFromEnemies) {
+    super(gameData, supportFromFriends, supportFromEnemies);
+  }
+
+  @Override
+  public StrengthAndRollCalculator getStrength() {
+    return new AaOffenseStrength(gameData, supportFromFriends, supportFromEnemies);
```

Comment:
There does not seem to be a lot going on in this class implementation. It looks like we could make `AaCombatValue` a concrete class by adding a boolean parameter for offense, defense, and adding a `Supplier<StrengthAndRollCalculator>` and then you could instantiate the base class and not need any inheritance.